packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -304,6 +353,20 @@ export interface AdvancedSecurityOptions {
    * @default - A Secrets Manager generated password
    */
   readonly masterUserPassword?: cdk.SecretValue;
+
+  /**
+   * True to enable SAML authentication for a domain.
+   *
+   * @see https://docs.aws.amazon.com/opensearch-service/latest/developerguide/saml.html
+   *
+   * @default - SAML authentication is disabled. Enabled if `samlAuthenticationOptions` is set.
+   */
+  readonly samlAuthenticationEnabled?: boolean;
+
+  /**
+   * Container for information about the SAML configuration for OpenSearch Dashboards.
+   */
+  readonly samlAuthenticationOptions?: SAMLOptionsProperty;
```

Comment:
Could you add a default stating "no SAML authentication options"? - `@default - no SAML authentication options`

I also think we should add a note here saying that if SAML authentication options are set, then `samlAuthenticationEnabled` will be enabled